Yank/sampling.py

```diff
@@ -789,7 +789,10 @@ def _initialize_netcdf(self):
 
     def _write_iteration_netcdf(self):
         super(ModifiedHamiltonianExchange, self)._write_iteration_netcdf()
-        if self.fully_interacting_state is not None:
+        if self.mpicomm:
+            # Only the root node will write data.
+            if self.mpicomm.rank != 0: return
```

Comment:
> Alternatively, autodetection could certainly go into a separate PR.

Oh! Sorry, that's what I meant. Now that I read what I wrote above, it's a mess. Anyway now I've merged my PR so you can fetch from upstream and start from there.

Maybe it would be useful to use branches instead of forks internally, so that anyone can modify the PR of somebody else.